src/python_testing/TC_IDM_1_4.py

```diff
@@ -145,13 +160,54 @@ async def steps_3_to_9(self, dummy_value):
 
-        self.print_step(6, "Skipping test until https://github.com/project-chip/connectedhomeip/issues/30991 resolved")
+        self.print_step(
+            6, "Verify DUT is able to responsed to InvokeRequestMessage that contains one paths InvokeRequest, and one InvokeRequest to unsupported endpoint")
+        # First finding non-existent endpoint
+        wildcard_descriptor = await dev_ctrl.ReadAttribute(dut_node_id, [(Clusters.Descriptor)])
+        endpoints = list(wildcard_descriptor.keys())
+        endpoints.sort()
+        non_existent_endpoint = next(i for i, e in enumerate(endpoints + [None]) if i != e)
 
-        self.print_step(7, "Skipping test until https://github.com/project-chip/connectedhomeip/issues/30986 resolved")
+        endpoint = 0
+        command = Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            Clusters.OperationalCredentials.Enums.CertificateChainTypeEnum.kDACCertificate)
+        invoke_request_1 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        endpoint = non_existent_endpoint
+        command = Clusters.GroupKeyManagement.Commands.KeySetRead(0)
+        invoke_request_2 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+        try:
+            result = await dev_ctrl.SendBatchCommands(dut_node_id, [invoke_request_1, invoke_request_2])
+            asserts.assert_true(type_matches(result, list), "Unexpected return from SendBatchCommands")
+            asserts.assert_equal(len(result), 2, "Unexpected number of InvokeResponses sent back from DUT")
+            asserts.assert_true(type_matches(
+                result[0], Clusters.OperationalCredentials.Commands.CertificateChainResponse), "Unexpected return type for first InvokeRequest")
+            asserts.assert_true(type_matches(
+                result[1], InteractionModelError), "Unexpected return type for second InvokeRequest")
+            asserts.assert_equal(result[1].status, Status.UnsupportedEndpoint,
+                                 "Unexpected Interaction model error, was expecting UnsupportedEndpoint")
+            logging.info(
+                "DUT successfully responded to first valid InvokeRequest, and successfully errored with UnsupportedEndpoint for the second")
+        except InteractionModelError:
+            asserts.fail("DUT failed to successfully responded to a InvokeRequest action with two valid commands")
+
+        self.print_step(7, "Verify DUT is able to responsed to InvokeRequestMessage that contains two valid paths. One of which requires timed invoke, and TimedRequest in InvokeResponseMessage set to true, but never sent preceding Timed Invoke Action")
+        endpoint = 0
+        command = Clusters.GroupKeyManagement.Commands.KeySetRead(0)
+        invoke_request_1 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        command = Clusters.AdministratorCommissioning.Commands.RevokeCommissioning()
```

Comment:
In test step 8 we send the exact same command where we get a cluster status saying that the commissioning window needs to be open. That would be a path-specific error/result. In step 7 we get a non-path error because the entire command failed